lib/kitchen/driver/ec2.rb

```diff
@@ -311,13 +319,16 @@ def copy_deprecated_configs(state)
         if config[:ssh_retries]
           state[:connection_retries] = config[:ssh_retries]
         end
-        if config[:username]
-          state[:username] = config[:username]
-        elsif instance.transport[:username] == instance.transport.class.defaults[:username]
-          # If the transport has the default username, copy it from amis.json
-          # This duplicated old behavior but I hate amis.json
-          ami_username = amis["usernames"][instance.platform.name]
-          state[:username] = ami_username if ami_username
+        state[:username] = config[:username]
+        unless instance.transport[:username] == instance.transport.class.defaults[:username]
+          state[:username] ||= instance.transport[:username]
+        end
+        if !state[:username]
+          if actual_platform
+            debug("No SSH username specified: using default username #{actual_platform.username} " \
```

Comment:
I feel like the right thing to do here is to have a better error message when SSH retries. It is total crap that we don't say why it's failing and what username we are using.